test/integration/expression/expression.test.ts

```diff
@@ -0,0 +1,183 @@
+import path from 'path';
+import fs from 'fs';
+import {globSync} from 'glob';
+import {
+    createPropertyExpression,
+    isFunction,
+    convertFunction,
+    toString,
+    ICanonicalTileID,
+    StylePropertyExpression
+} from '../../../src/index';
+import ExpressionParsingError from '../../../src/expression/parsing_error';
+import {Result} from '../../../src/util/result';
+import {getGeometry} from './lib/geometry';
+import {stringify} from './lib/util';
+import {deepEqual, stripPrecision} from './lib/json-diff';
+
+const DECIMAL_SIGNIFICANT_FIGURES =  6;
+
+type ExpressionFixture = {
+    propertySpec: any;
+    expression: any[];
+    inputs:any[];
+    expected: {
+        compiled?: {
+            result?: any;
+            isFeatureConstant?: any;
+            isZoomConstant?: any;
+            type?: any;
+        };
+        outputs? : any;
+        serialized?: any;
+    };
+}
+
+const expressionTestFileNames = globSync('**/test.json', {cwd: __dirname});
+describe('expression', () => {
+
+    for (const expressionTestFileName of expressionTestFileNames) {
+        test(expressionTestFileName, () => {
+
+            const fixture = JSON.parse(fs.readFileSync(path.join(__dirname, expressionTestFileName), 'utf8'));
+
+            const result = evaluateFixture(fixture);
+
+            if (process.env.UPDATE) {
```

Comment:
Needs some documentation maybe.

Reply:
Good point, I'll look to see if this is mentioned somewhere.

Reply:
Done in the contributing guide.